File: medplum-java/medplum-model/src/main/java/com/fhir/Parameters.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableParameters.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Parameters {
  String resourceType();

  Optional<uri> implicitRules();

  Optional<id> id();

  Optional<Meta> meta();

  Optional<code> language();

  Optional<List<Parameters_Parameter>> parameter();

  static ImmutableParameters.ResourceTypeBuildStage builder() {
    return ImmutableParameters.builder();
  }
}

File: medplum-typescript/src/model/fhir/Parameters.ts
import { code } from "./code";
import { id } from "./id";
import { Meta } from "./Meta";
import { Parameters_Parameter } from "./Parameters_Parameter";
import { uri } from "./uri";

export interface Parameters {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    parameter: Parameters_Parameter[] | null | undefined;
}
